cv32e40s/tests/programs/custom/illegal_access_loop_tests/illegal_access_loop_tests.c

```diff
@@ -0,0 +1,89 @@
+/*
+**
+** Copyright 2022 OpenHW Group
+**
+** Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+** https://solderpad.org/licenses/
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+*******************************************************************************
+**
+** Contains generated tests on U-mode access and privilege instructions.
+**
+*******************************************************************************
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "corev_uvmt.h"
+#include <stdint.h>
+
+// extern and global variable declaration
+extern volatile void  setup_pmp();
+extern volatile unsigned int illegal_full(), csr_loop(), mco_loop(), Check_mcounteren();
+
+
+// Declaration of assert 
+static __inline__ void assert_or_die(uint32_t actual, uint32_t expect, char *msg) {
+  if (actual != expect) {
+    printf(msg);
+    printf("expected = 0x%lx (%ld), got = 0x%lx (%ld)\n", expect, (int32_t)expect, actual, (int32_t)actual);
+    exit(EXIT_FAILURE);
+  }
+}
+
+void change_m_mode(void) {
+  //s4 is x20
+  __asm__ volatile("addi s4, x0, 0x3"); // add 3 to the s4 register, which will be read by the trap handler late
+  __asm__ volatile("ecall");
+  __asm__ volatile("li s4, 0"); // Reset the value after entering machine mode
+}
+
+void illegal_custom_loop(void) {
+    /* 
+    Tests U-mode access to various custom functions which are not yet implemented. Should all trap.
```

Comment:
What are "custom functions"?

And, from "not yet implemented" I wonder, are these expected to be implemented soon?
If the spec say we should have some functionality, then the tests should expect them to work, and either error out on parts that aren't implemented or we can leave a TODO comment to enable them once the RTL is ready for it.

Reply:
Vplan says this:

Feature descr: ""The subspace of the SYSTEM major opcode shown in Figure 3.30 is designated for custom use."

"Unprivileged or User-Level"

(40s has no such thing.)"

Verification Goal: "Execute custom SYSTEM "Unprivileged or User-Level" instructions, ensure they are illegal instructions."

Reply:
Ok, it refers to custom instructions. Then it's fine.